### ISSUE TITLE
fix AssetAliasUsed issue

### ISSUE THREAD

**Issue by rmosolgo** [rails/sprockets-rails#257]
In `react-rails`, we add different paths to the asset pipeline based on the app's config. With Sprockets 2 + sprockets-rails 2.3.1, this causes `AssetAliasUsed` error. 

Is this error actually a problem? If not, it would be nice to remove it (or fix the check so it isn't raised in this case) and ship 2.3.2. If it _is_ a problem, what does it mean? Or how should I fix it?

See [full conversation](https://github.com/rails/sprockets-rails/issues/247#issuecomment-103162751), here's my initial report copied from that thread:

## react-rails setup 

In `react-rails`, app config specifies either `production` or `development`, based on that, we add a path to `app.config.assets.paths`, roughly speaking:

```ruby
# add directory to asset pipeline, 
# based on the app config
app.config.assets.paths << root_path.join('lib/assets/react-source/').join(directory).to_s
```

([full src](https://github.com/reactjs/react-rails/blob/master/lib/react/rails/railtie.rb#L45-L49))

We do this so that the app only needs `//= react`, then, each environment gets its version of React.js (the development version has nice error messages, the production version has none). 

## error 

However, we get an `AssetAliasUsed` from the `javascript_include_tag` helper:

![image](https://cloud.githubusercontent.com/assets/2231765/7687233/8fe427e4-fd50-11e4-978e-f38b116e53d2.png)

Sure enough, these don't match: 

![image](https://cloud.githubusercontent.com/assets/2231765/7687214/69c605fa-fd50-11e4-870b-00b8114daa7d.png)

Interestingly, if I upgrade to Sprockets 3 + sprockets-rails 2.3.1, I don't get this error (which is fine for me, but not fine for many other `react-rails` users...).

Is this the expected outcome? Are we misusing the asset pipeline? Anything else I can do to diagnose or address this?

We're really hoping to support an API where `//= require react` can provide different versions of the file based on environment configs. Previously we copied JS files into local directories, but that had its own issues!

Thanks!

**Comment by cseydel** [comment]
I can confirm this: ``Interestingly, if I upgrade to Sprockets 3 + sprockets-rails 2.3.1, I don't get this error``

**Comment by rafaelfranca** [comment]
This should be fixed on sprockets-rails 2.3.2. Please give it a try

**Comment by aprescott** [comment]
@rafaelfranca thank you!
